src/Services/CheckoutService.php

```diff
@@ -1,251 +0,0 @@
-<?php
-
-declare(strict_types=1);
-
-namespace Tipoff\Checkout\Services;
-
-use Illuminate\Database\Eloquent\Model;
-use Tipoff\Checkout\Models\Cart;
-
-class CheckoutService
-{
-    /* Tax & fee application methods */
-    const APPLICATION_ORDER = 'order';
-    const APPLICATION_EACH = 'each';
-    const APPLICATION_PRODUCT = 'product';
-    const APPLICATION_BOOKING = 'booking';
-    const APPLICATION_PARTICIPANT = 'participant';
-
-    /**
-     * Check if discount/voucher code is valid.
-     *
-     * @param string $code
-     * @return mixed
-     * @psalm-suppress UndefinedMethod
-     */
-    public function findDeductionByCode($code)
-    {
-        if (strlen($code) < 4 && strlen($code) > 16) {
-            return false;
-        }
-
-        if ($voucher = app('voucher')::validAt()->where('code', $code)->first()) {
-            return $voucher;
-        }
-
-        if ($discount = app('discount')::available()->where('code', $code)->first()) {
-            return $discount;
-        }
-
-        return false;
-    }
-
-    /**
-     * Apply discount or voucher code to cart.
-     *
-     * @param string $code
-     * @return Cart
-     */
-    public function applyCodeToCart($code, $cart)
-    {
-        $deduction = $this->findDeductionByCode($code);
-
-        if (empty($deduction)) {
-            throw new \Exception("Code {$code} is invalid.");
-        }
-
-        $cart->applyDeduction($deduction);
```

Comment:
Leverages common `CartDeduction` interface to change this into `$deduction->applyToCart($this)`